indexing-files-with-python.mdx

```diff
@@ -0,0 +1,443 @@
+2. Navigate to the directory: `cd xtools/xfileindex`
+3. Install the required packages using pip: `pip install -r requirements.txt`
+The script uses some extra packages (*xata,PyPDF2*) which are not typically available with the default Python installed in your system, so we need to make sure they’re installed before we proceed.
+4. Set your Xata API key as an environment variable on your terminal. You can generate an API key within the [account settings](https://app.xata.io/settings) page. Alternatively, you can use [a .env file](https://xata.io/docs/sdk/python/overview#dotenv), to store the API key.
+
+```bash
+export XATA_API_KEY="xau_mykey”
+```
+
+1. We need to gather a few parameters about our database and schema:
+- The database endpoint url. You can find that in the Settings page of your database, under “Database endpoint”. 
+- The table name where our files are stored.
+- The name(s)s of the column(s) where our files are stored
+2. Knowing all the above, we can run the script as follows:
+
+```bash
+python xfileindex.py --db https://{workspace}.{region}.xata.sh/db/{db} --table tablename --columns firstcolumn,secondcolumn
+```
+
+Example output:
+
+```bash
+python xfileindex.py --db https://ws-fdrujb.eu-central-1.xata.sh/db/myfilesdb --table filestable --columns firstcolumn,secondcolumn
+Created new table filestableIndex
+
+Downloading file from table: filestable , record: rec_cjmvbvr9o1hng2274a9g , column: firstcolumn , filename: lotr.pdf
+- Processing record: rec_cjmvbvr9o1hng2274a9g , column: firstcolumn , filename: lotr.pdf , type: application/pdf in 1199 chunks:
+  Indexed 100 / 1199 chunks.
+  Indexed 200 / 1199 chunks.
+  Indexed 300 / 1199 chunks.
+  Indexed 400 / 1199 chunks.
+  Indexed 500 / 1199 chunks.
+  Indexed 600 / 1199 chunks.
+  Indexed 700 / 1199 chunks.
+  Indexed 800 / 1199 chunks.
+  Indexed 900 / 1199 chunks.
+  Indexed 1000 / 1199 chunks.
+  Indexed 1100 / 1199 chunks.
+  Indexed 1199 / 1199 chunks.
+
+Downloading file from table: filestable , record: rec_cjmscqj9n6h19i0nbdng , column: secondcolumn , filename: sample-2mb-text-file.txt
+- Processing record: rec_cjmscqj9n6h19i0nbdng , column: secondcolumn , filename: sample-2mb-text-file.txt , type: text/plain in 11 chunks:
+  Indexed 11 / 11 chunks.
+
+Downloading file from table: filestable , record: rec_cjmscqj9n6h19i0nbdng , column: secondcolumn , filename: mycsv.csv
+- Processing record: rec_cjmscqj9n6h19i0nbdng , column: secondcolumn , filename: mycsv.csv , type: text/csv in 1 chunk:
+  Indexed 1 / 1 chunk.
+```
+
+### Under the hood
+
+Let’s take a moment to review the output and dive deeper into how things worked here.
+
+The script created a new table filestableIndex, by appending “Index” to our source table name. Alternatively you can specify your preferred name for the target table by providing the `dest`parameter (`--dest customTableName`). Also, in case your table lives in a branch other than `main`, you can add the branch parameter (`--branch dev`).
```

Comment:
```suggestion
The script created a new table `filestableIndex`, by appending "Index" to our source table name. Alternatively you can specify your preferred name for the target table by providing the `dest` parameter (`--dest customTableName`). Also, in case your table lives in a branch other than `main`, you can add the branch parameter (`--branch dev`).
```